src/NexusMods.App.UI/Pages/LibraryPage/LibraryViewModel.cs

```diff
@@ -167,10 +168,19 @@ public LibraryViewModel(
             Adapter.MessageSubject.SubscribeAwait(
                 onNextAsync: async (message, cancellationToken) =>
                 {
-                    foreach (var id in message.Ids)
+                    if (message.Payload.TryPickT0(out var multipleIds, out var singleId))
                     {
-                        var libraryItem = LibraryItem.Load(_connection.Db, id);
-                        if (!libraryItem.IsValid()) continue;
+                        foreach (var id in multipleIds)
+                        {
+                            var libraryItem = LibraryItem.Load(_connection.Db, id);
+                            if (!libraryItem.IsValid()) continue;
+                            await InstallLibraryItem(libraryItem, _loadout, cancellationToken);
+                        }
+                    }
+                    else
+                    {
+                        var libraryItem = LibraryItem.Load(_connection.Db, singleId);
+                        if (!libraryItem.IsValid()) return;
                         await InstallLibraryItem(libraryItem, _loadout, cancellationToken);
```

Comment:
Was there no way to just always pass a collection, and just pas a collection of one item in the cases of a single item? 
Or is there an advantage of keeping the cases separate?

Reply:
It just looked weird to return an array with a single item. Now that we're using composition, we can have these distinctions.